article/app/controllers/ArticleController.scala

```diff
@@ -234,15 +260,29 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 }
 
 object ParseBlockId extends RegexParsers {
-  def apply(input: String): Option[String] = {
-    def withParser: Parser[Unit] = "with:" ^^ { _ => () }
-    def block: Parser[Unit] = "block-" ^^ { _ => () }
-    def id: Parser[String] = "[a-zA-Z0-9]+".r
-    def expr: Parser[String] = withParser ~> block ~> id
+
```

Comment:
👍 to parboiled!
[We've use it on the grid too](https://github.com/guardian/grid/blob/master/media-api/app/lib/querysyntax/QuerySyntax.scala) (perhaps some refinements needed)